File: Chapter 03/C#/3.3.4 ISP/After/Midfielder.cs
using System;

namespace Chapter03.ISP.After;

public class Midfielder : IFieldPlayerTraining
{
    public void PracticeShooting() 
    { 
        Console.WriteLine("  [Midfielder] Practicing shooting drills."); 
    }
    
    public void PracticeTackling() 
    { 
        Console.WriteLine("  [Midfielder] Practicing slide tackles."); 
    }
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.3.4 ISP/After/Midfielder.java
package com.grokkingsoftwarearchitecture.chapter03.isp.after;

public class Midfielder implements FieldPlayerTraining {
    public void practiceShooting() {
        System.out.println("  [Midfielder] Practicing shooting drills.");
    }

    public void practiceTackling() {
        System.out.println("  [Midfielder] Practicing slide tackles.");
    }
}